python-frontend/src/test/java/org/sonar/python/semantic/SymbolTableBuilderTest.java

```diff
@@ -285,6 +288,155 @@ public void importing_stdlib() {
+    SymbolImpl datastructures = (SymbolImpl) childrenSymbolByName.get("datastructures");
+    assertThat(datastructures.getChildrenSymbolByName().values()).extracting(Symbol::name).contains("csp");
+    assertThat(datastructures.getChildrenSymbolByName().values()).extracting(Symbol::name).doesNotContain("Headers");
+
+    CallExpression callExpression = (CallExpression) ((ExpressionStatement) functionDef.body().statements().get(2)).expressions().get(0);
+    Symbol qualifiedExpressionSymbol = callExpression.calleeSymbol();
+    assertThat(qualifiedExpressionSymbol.is(Symbol.Kind.CLASS)).isTrue();
+    assertThat(qualifiedExpressionSymbol.fullyQualifiedName()).isEqualTo("werkzeug.datastructures.csp.ContentSecurityPolicy");
+  }
+
+  @Test
+  public void importing_parent_after_submodule_2() {
+    FunctionDef functionDef = functionTreesByName.get("importing_parent_after_submodule_2");
+    Map<String, Symbol> symbolByName = getSymbolByName(functionDef);
+
+    assertThat(symbolByName).containsOnlyKeys("werkzeug");
+    SymbolImpl werkzeug = (SymbolImpl) symbolByName.get("werkzeug");
+    assertThat(werkzeug.usages()).extracting(Usage::kind).containsExactly(Usage.Kind.IMPORT, Usage.Kind.IMPORT, Usage.Kind.IMPORT, Usage.Kind.OTHER);
+    Map<String, Symbol> childrenSymbolByName = werkzeug.getChildrenSymbolByName();
+    assertThat(childrenSymbolByName.values()).extracting(Symbol::name).contains("datastructures");
+    SymbolImpl datastructures = (SymbolImpl) childrenSymbolByName.get("datastructures");
+    assertThat(datastructures.getChildrenSymbolByName().values()).extracting(Symbol::name).contains("Headers", "csp");
+
+    CallExpression callExpression = (CallExpression) ((ExpressionStatement) functionDef.body().statements().get(3)).expressions().get(0);
+    Symbol qualifiedExpressionSymbol = callExpression.calleeSymbol();
+    assertThat(qualifiedExpressionSymbol.is(Symbol.Kind.CLASS)).isTrue();
+    assertThat(qualifiedExpressionSymbol.fullyQualifiedName()).isEqualTo("werkzeug.datastructures.csp.ContentSecurityPolicy");
+  }
+
+  @Test
+  public void importing_submodule_twice() {
+    FunctionDef functionDef = functionTreesByName.get("importing_submodule_twice");
+    Map<String, Symbol> symbolByName = getSymbolByName(functionDef);
+
+    assertThat(symbolByName).containsOnlyKeys("werkzeug");
+    SymbolImpl werkzeug = (SymbolImpl) symbolByName.get("werkzeug");
+    assertThat(werkzeug.usages()).extracting(Usage::kind).containsExactly(Usage.Kind.IMPORT, Usage.Kind.IMPORT, Usage.Kind.OTHER);
+    Map<String, Symbol> childrenSymbolByName = werkzeug.getChildrenSymbolByName();
+    assertThat(childrenSymbolByName.values()).extracting(Symbol::name).containsExactly("datastructures");
+    SymbolImpl datastructures = (SymbolImpl) childrenSymbolByName.get("datastructures");
+    assertThat(datastructures.getChildrenSymbolByName().values()).extracting(Symbol::name).contains("Headers");
```

Comment:
Would it make sense to check that we do not have duplicates in the children symbols?